saleor/app/tasks.py

```diff
@@ -46,3 +51,58 @@ def install_app_task(job_id, activate=False):
         app_installation.message = "Unknown error. Contact with app support."
     app_installation.status = JobStatus.FAILED
     app_installation.save(update_fields=["message", "status"])
+
+
+def _raw_remove_deliveries(deliveries_ids):
+    deliveries = EventDelivery.objects.filter(id__in=deliveries_ids)
+    payloads_ids = list(
+        EventPayload.objects.filter(
+            Exists(deliveries.filter(payload_id=OuterRef("id")))
+        ).values_list("id", flat=True)
+    )
+    payloads = EventPayload.objects.filter(id__in=payloads_ids)
```

Comment:
Why are you fetching `payload_ids` and then fetching those payloads?
Cannot just do:
```suggestion
    payloads = EventPayload.objects.filter(
        Exists(deliveries.filter(payload_id=OuterRef("id")))
    )

```